include/NAS2D/StringUtils.h

```diff
@@ -17,6 +17,44 @@
 
 namespace NAS2D
 {
+	template<typename T>
+	T from_string(const std::string& value)
+	{
+		return T(value);
+	}
+
+	template<>
+	std::string from_string(const std::string& value);

```

Comment:
Curious why this wasn't implemented inline. Usually template methods have the body in the header file. I guess with a specialization like this, it is possible to implement in a source file. That would prevent inlining though, and these are pretty small and easily inlined methods.

Reply:
Putting explicit template specializations in the header causes VS to throw an "already defined" error:
```
1>E:\Git\nas2d-core\src\StringUtils.cpp(22,1): error C2766: explicit specialization; 'std::string NAS2D::from_string<std::string>(const std::string &)' has already been defined
1>E:\Git\nas2d-core\include\NAS2D\StringUtils.h(27): message : see previous definition of 'from_string'
```

Reply:
Defining explicit template specializations in the header causes the VS linker to freak out:

```
2>Filesystem.test.obj : error LNK2005: "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl NAS2D::from_string<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (??$from_string@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@NAS2D@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABV12@@Z) already defined in DataUtils.test.obj
2>StringUtils.test.obj : error LNK2005: "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl NAS2D::from_string<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (??$from_string@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@NAS2D@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABV12@@Z) already defined in DataUtils.test.obj
2>NAS2D.lib(StringUtils.obj) : warning LNK4075: ignoring '/EDITANDCONTINUE' due to '/SAFESEH' specification
2>NAS2D.lib(StringUtils.obj) : error LNK2005: "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl NAS2D::from_string<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (??$from_string@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@NAS2D@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABV12@@Z) already defined in DataUtils.test.obj
2>NAS2D.lib(Image.obj) : error LNK2005: "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl NAS2D::from_string<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (??$from_string@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@NAS2D@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABV12@@Z) already defined in DataUtils.test.obj
2>NAS2D.lib(Filesystem.obj) : error LNK2005: "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl NAS2D::from_string<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (??$from_string@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@NAS2D@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABV12@@Z) already defined in DataUtils.test.obj
2>E:\Git\nas2d-core\test\x86\Debug\test.exe : fatal error LNK1169: one or more multiply defined symbols found

```